Documentation/gettingstarted/external-workloads.rst

```diff
@@ -0,0 +1,353 @@
+
+    If not using a pre-set IP, get the service IP with:
+
+    .. code:: bash
+
+      kubectl -n kube-system get svc clustermesh-apiserver -o jsonpath='{.status.loadBalancer.ingress[0].ip}{"\n"}'
+
+  .. group-tab:: AWS
+
+    .. parsed-literal::
+
+      --set clustermesh.apiserver.service.type=LoadBalancer
+      --set clustermesh.apiserver.service.annotations."service\.beta\.kubernetes\.io/aws-load-balancer-internal"="true"
+
+    If not using a pre-set IP, get the service IP with:
+
+    .. code:: bash
+
+      kubectl -n kube-system get svc clustermesh-apiserver -o jsonpath='{.status.loadBalancer.ingress[0].ip}{"\n"}'
+
+.. note::
+
+   Make sure that you use the namespace in which cilium is
+   running. Depending on which installation method you chose, this
+   could be ``kube-system`` or ``cilium``.
+
+Tell your cluster about external workloads
+==========================================
+
+To allow an external workload to join your cluster, the cluster must
+be informed about each such workload. This is done by adding a
+``CiliumExternalWorkload`` (CEW) CRD for each external workload. CEW
+CRD specifies the name, namespace, and labels for the workload. For
+now you must also allocate a small IP CIDR that must be unique to the
+workload. For example, for a VM named ``runtime`` that is to join the
+``default`` namespace, you could create a file ``runtime.yaml`` with
+the following contents:
+
+.. parsed-literal ::
+
+    apiVersion: cilium.io/v2
+    kind: CiliumExternalWorkload
+    metadata:
+      name: runtime
+      namespace: default
+      labels:
+        app: runtime
+    spec:
+      ipv4-alloc-cidr: 10.192.1.0/30
+
+Apply this with:
+
+.. code:: bash
+
+    kubectl apply -f runtime.yaml
+
+Extract the TLS keys for external workloads
+===========================================
+
+Cilium control plane performs TLS based authentication and encryption.
+For this purpose, the TLS keys and certificates of
+``clustermesh-apiserver`` need to be made available to external
+workload that wish to join the cluster.
+
+Extract the TLS certs from your k8s cluster:
+
+.. code:: bash
+
+    ../../contrib/k8s/extract-external-workload-certs.sh
+
+This saves the certs (``ca.crt``, ``tls.crt``, ``tls.key``) to the
+current directory. These files need to be copied to your external
+workload.
+
+Install and configure Cilium on external workloads
+##################################################
+
+Log in to the external workload. First make sure the hostname matches
+the name used in the CiliumExternalWorkload CRD:
+
+.. code:: bash
+
+    hostname
+
+By now you should be able to find the corresponding CRD in your k8s
+cluster (``<name>`` is the output from ``hostname`` above):
+
+.. code:: bash
+
+    kubectl get cew <name>
+
+Next, copy the saved TLS certs from your kubectl command line and
+``contrib/k8s/install-external-workload.sh`` from the Cilium release
+you downloaded in the beginning to the external workload. Then,
```

Comment:
I didn't see an explicit command to download this file, and note that branch versions of the docs will use `helm install ...`, they will *not* download a Cilium tarball.

I suggest we add a `.. parsed-literal::` `curl` command to pull the install-external-workload.sh script from the current branch.

Reply:
Missed that difference, changed to download the script from the master branch.